sdk/rust/oak/tests/handle_extract_inject.rs

```diff
@@ -0,0 +1,182 @@
+include!(concat!(env!("OUT_DIR"), "/handle_tests/tests.rs"));
+
+use oak::{
+    handle::{extract_handles, inject_handles},
+    io::{Receiver, Sender},
+};
+
+#[test]
+fn extract_struct() {
+    let mut message = TestMessage {
+        other_arbitrary_field: "Test".to_string(),
+        test_sender: Some(sender(42)),
+        test_receiver: Some(receiver(1337)),
+    };
+
+    let handles = extract_handles(&mut message);
+
+    assert_eq!(handles, vec![42, 1337]);
+    assert_eq!(
+        message,
+        TestMessage {
+            other_arbitrary_field: "Test".to_string(),
+            test_sender: Some(sender(0)),
+            test_receiver: Some(receiver(0)),
+        }
+    );
+}
+
+#[test]
+fn enum_extract_sender() {
+    let mut message = TestMessageWithEnum {
+        either: Some(test_message_with_enum::Either::EitherSender(sender(42))),
+    };
+
+    let handles = extract_handles(&mut message);
+
+    assert_eq!(handles, vec![42]);
+    assert_eq!(
+        message,
+        TestMessageWithEnum {
+            either: Some(test_message_with_enum::Either::EitherSender(sender(0))),
+        }
+    );
+}
+
+#[test]
+fn enum_extract_receiver() {
+    let mut message = TestMessageWithEnum {
+        either: Some(test_message_with_enum::Either::EitherReceiver(receiver(42))),
+    };
+
+    let handles = extract_handles(&mut message);
+
+    assert_eq!(handles, vec![42]);
+    assert_eq!(
+        message,
+        TestMessageWithEnum {
+            either: Some(test_message_with_enum::Either::EitherReceiver(receiver(0))),
+        }
+    );
+}
+
+#[test]
+fn enum_inject_sender() {
```

Comment:
Also a test case in which we attempt to inject more handles, and one in which we attempt to inject fewer handles?

Reply:
Done.